pkg/controller/bastion/actuator_reconcile.go

```diff
@@ -108,90 +107,100 @@ func (a *actuator) Reconcile(ctx context.Context, bastion *extensionsv1alpha1.Ba
 }
 
 func getPrivateIPAddress(nic *network.Interface) (string, error) {
-	ipConfigurations := *nic.InterfacePropertiesFormat.IPConfigurations
-	if len(ipConfigurations) > 0 {
+	ipConfigurations := *nic.IPConfigurations
+	if len(ipConfigurations) != 0 && len(ipConfigurations) > 0 {
```

Comment:
to check if nic is nil was set up in `ensureNic` which used to return nic object.  adding a check, in case if not already ensured by the caller. 
dfaab642

Reply:
```go
	if nic == nil && *nic.IPConfigurations == nil {
		return "", fmt.Errorf("nic %s is nil", *nic.ID)
	}

	ipConfigurations := *nic.IPConfigurations
```

@tedteng now you made two mistakes with your change
1. You probably want to run into the if-case if either nic OR *nic.IPConfigurations is nil, otherwise you have a nil pointer dereference with the line following the if-case (`ipConfigurations := *nic.IPConfigurations`)  -> change the AND operator to OR
2. Once you apply the change from 1. you can't access *nic.ID within the if-case if nic itself is nil.

Reply:
fixed https://github.com/gardener/gardener-extension-provider-azure/pull/397/commits/2d1c47093acfed887ea5315bd49fe759bee0d92b
I removed the check for `nic == nil` because it's assumable that it's not nil at this point. Upper on the stack we check if get any errors while querying the `nic`.
I mean otherwise, we have to add this kind of check in every function where we pass the pointer. It will make our code insane. So if there no error we assume that the structure is not nil.